### DISCUSSION TITLE
publish MQTT syntax

### DISCUSSION THREAD

**Discussion by Avatarchik** [uNetworking/uWebSockets.js#513] [Q&A]
Hi! This example does not work for me, the server will not send to all subscribers

```
          if (json.type == "pc") {
                    ws.subscribe(json.id +"/pc");
                } else if (json.type == "mobile") {
                    ws.subscribe(json.id +"/mobile");
               }
                 ws.publish(json.id, JSON.stringify(msg));
```

or do I need to do this?

```
          if (json.type == "pc") {
                    ws.subscribe(json.id +"/pc");
                     ws.subscribe(json.id);
                } else if (json.type == "mobile") {
                    ws.subscribe(json.id +"/mobile");
                     ws.subscribe(json.id);
               }
                 ws.publish(json.id, JSON.stringify(msg));

```

**Comment by obliwetator** [comment]
I've got the same issue after 2 hours I managed to find why. The first time you subscribe is ignored for some reason. If you subscribe ws.subscribe("1") <--- this will be ignored for publish but ws.subscribe("2") <---- this will publish to all clients. This started  to happen after I upgraded to node.js 16 and the latest 19.2 version. I don't know if it's an issue with the library or node.js
If someone could move it back as a bug or try to investigate further

**Comment by hst-m** [comment]
this works fine so you should post your code if you need help @ws295203 
```javascript
app=require('uWebSockets.js').App().ws('/',{
   open:ws=>{ws.subscribe('1');app.publish('1','a')}
}).listen(80,s=>{})
```
```javascript
new WebSocket('ws://localhost')
```

the OP @Avatarchik does not understand the MQTT syntax and apparently is not able to run the code to try themselves, their 2nd example seems to be what they are trying to do but dont know if they tried it

**Reply by hst-m** [reply]
@Avatarchik v19 you need to use `app.publish` not `ws.publish`, ws.publish does not send to same ws

**Reply by obliwetator** [reply]
Yea it works now when i use app.publish. The example didn't use that so I guess it's kind of strange it worked fine up until I updated. Thanks for the response @hst-m

**Reply by hst-m** [reply]
The docs do need to be updated in multiple places, the only place to find info on changes currently is https://github.com/uNetworking/uWebSockets/releases
